core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -2834,6 +2834,42 @@ public static ByteString bitAnd(ByteString b0, ByteString b1) {
     return binaryOperator(b0, b1, (x, y) -> (byte) (x & y));
   }
 
+  /** Helper function for implementing <code>BITCOUNT</code>. Counts the number
+   * of bits set in an integer value. */
+  public static long bitCount(long b) {
+    return Long.bitCount(b);
+  }
+
+  private static final BigDecimal BITCOUNT_MAX = new BigDecimal("18446744073709551615");
```

Comment:
Why this number？

Reply:
This is `2^64 - 1`. I found it through testing in MySQL. I added a comment to say that it is `2^64 - 1`.

Reply:
Maybe we can use 
```
Math.pow(2,64) -1
```
generate it?

Reply:
BigDecimal has s shiftleft method

Reply:
BigInteger has a shiftleft, BigDecimal does not. I have updated the constants to use BigDecimal power, subtract and negate.